### PR TITLE
[BUGFIX beta] Don't mask errors in commit

### PR DESCRIPTION
**Pull request by chancancode**

### CODE REVIEW

File: packages/@ember/-internals/glimmer/lib/environment.ts [hunk 1]
```diff
@@ -99,6 +99,12 @@ export default class Environment extends GlimmerEnvironment {
 
     try {
       super.commit();
+    } catch(error) {
+      if (completedWithoutError) {
+        throw error;
+      } else {
+        console.error(error);
```
Comment (pzuraq) [inline packages/@ember/-internals/glimmer/lib/environment.ts]: So, if I'm reading this correctly, if `completedWithoutError` is true then `error` comes from the commit itself. What is the `error` if `completedWithoutError` is false? Is it the error of the outer `try/catch`?

Comment (chancancode) [reply packages/@ember/-internals/glimmer/lib/environment.ts]: Correct. Basically we were already planning to throw a different error from where we are called.

Comment (pzuraq) [reply packages/@ember/-internals/glimmer/lib/environment.ts]: Ahhh I see now, ok 👍

**Review by pzuraq — APPROVED** [review]

**Review by rwjblue — CHANGES_REQUESTED** [review]
I discussed this with @chancancode a bit at the F2F. This change concerns me a bit, specifically that the debugging experience would suffer (since you will now not get an error inside your component's `didInsertElement` or other userland hook location because we try/catch here I think that chrome would stop here **deep** in the renderer internals).

This can be somewhat mitigated by only adding the `try`/`catch` when `completedWithoutError` is `false` in the Environment.prototype.commit.

**Comment by locks** [comment]
@rwjblue @chancancode is this path being abandoned?

**Comment by chancancode** [comment]
Pretty sure we did this already

**Pull request closed without merging**